Frameworks/Foundation/NSOperation.mm

```diff
@@ -40,13 +40,13 @@
 #import <mutex>
 
 @implementation NSOperation {
-    StrongId<NSMutableArray> _dependencies;
+    StrongId<NSMutableSet> _dependencies;
+    NSUInteger _outstandingDependenciesCount;
     StrongId<void(^)()> _completionBlock;
     std::condition_variable_any _finishCondition;
     std::recursive_mutex _finishLock;
     std::recursive_mutex _dependenciesLock;
     std::recursive_mutex _completionBlockLock;
```

Comment:
this is a terrifying number of locks for a single object. Can you add a comment here explaining what order locks need to be taken in?